maven-core/src/main/java/org/apache/maven/toolchain/building/DefaultToolchainsBuilder.java

```diff
@@ -49,7 +56,9 @@
     implements ToolchainsBuilder
 {
     private MavenToolchainMerger toolchainsMerger = new MavenToolchainMerger();
-    
+
```

Comment:
DefaultToolchainsWriter is annotated with @Component, so it is injectable here. Please replace new instance creation with @Inject (if DefaultSettingsBuilder does the same, we'll need to fix it there too)

Reply:
I changed it to be injected via constructor injection, since that enables us to test against a real object of the writer in the unit tests. Alternatively we can mark the `ToolchainsWriter` as a spy object, but I do not prefer that since we do not intend to stub or verify methods.
As the `DefaultSettingsBuilder` does not have a lot of tests, field injection is used there without problems.

Reply:
Maven code uses field injection by default, not sure if constructor injection will work, it all depends on sisu/guice.

Reply:
The javadoc suggests so: 

> The class being constructed must have exactly one of its
 *     constructors marked with {@code @Inject} or must have a constructor taking no parameters

https://github.com/sonatype/sisu-guice/blob/master/core/src/com/google/inject/Inject.java

But for the sake of consistency I can go for the Spy object with field injection. What do you prefer?

Reply:
I like consistency :) And Maven doesn't use classpath scanning at runtime. There's a descriptor generated at compile time with that kind of information (to have a shorter warmup at runtime)  and I don't think that descriptor supports constructor injection.